cargo-shuttle/src/lib.rs

```diff
@@ -377,4 +428,131 @@ mod tests {
+                    .unwrap(),
+                &test_version,
+            )
+            .unwrap();
+            Shuttle::check_lib_version(
+                Document::from_str("[dependencies.shuttle-service]\nversion = \"0.4.1\"").unwrap(),
+                &test_version,
+            )
+            .unwrap();
+        }
+
+        #[test]
+        fn invalid_versions() {
+            Shuttle::check_lib_version(
+                Document::from_str("[dependencies]\nshuttle-service = \"0.3.1\"").unwrap(),
+                &Version::parse("0.4.1").unwrap(),
+            )
+            .unwrap_err();
+
+            Shuttle::check_lib_version(
+                Document::from_str("[dependencies]\nshuttle-service = \"0.3.1\"").unwrap(),
+                &Version::parse("1").unwrap(),
+            )
+            .unwrap_err();
+
+            Shuttle::check_lib_version(
+                Document::from_str("[dependencies]\nshuttle-service = \"0.3.1\"").unwrap(),
+                &Version::parse("0.4").unwrap(),
+            )
+            .unwrap_err();
+        }
+
+        #[test]
+        fn valid_versions() {
+            let version_four = Version::parse("0.4").unwrap();
+
+            Shuttle::check_lib_version(
+                Document::from_str("[dependencies]\nshuttle-service = \"0.4\"").unwrap(),
+                &version_four,
+            )
+            .unwrap();
+
+            Shuttle::check_lib_version(
+                Document::from_str("[dependencies]\nshuttle-service = \"0.4.0\"").unwrap(),
+                &version_four,
+            )
+            .unwrap();
+
+            Shuttle::check_lib_version(
+                Document::from_str("[dependencies]\nshuttle-service = \"0.4.5\"").unwrap(),
+                &version_four,
+            )
+            .unwrap();
+
+            let version_four_point_one = Version::parse("0.4.3").unwrap();
+
+            Shuttle::check_lib_version(
+                Document::from_str("[dependencies]\nshuttle-service = \"0.4\"").unwrap(),
+                &version_four_point_one,
+            )
+            .unwrap();
+
+            Shuttle::check_lib_version(
+                Document::from_str("[dependencies]\nshuttle-service = \"0.4.1\"").unwrap(),
+                &version_four_point_one,
+            )
+            .unwrap();
+
+            Shuttle::check_lib_version(
+                Document::from_str("[dependencies]\nshuttle-service = \"0.4.5\"").unwrap(),
+                &version_four_point_one,
+            )
+            .unwrap();
+
+            Shuttle::check_lib_version(
+                Document::from_str("[dependencies]\nshuttle-service = \"0.5\"").unwrap(),
+                &version_four_point_one,
+            )
+            .unwrap();
+
+            Shuttle::check_lib_version(
+                Document::from_str("[dependencies]\nshuttle-service = \"2\"").unwrap(),
+                &version_four_point_one,
+            )
+            .unwrap();
```

Comment:
Hmm, should these be vaild? These are behind the server and we have no backward compatibility at this time

Reply:
Yeah the tests are a little complicated to grapple. I think it says that if the clients shuttle_service is version 0.5 or 2 then it works on a server *running* 0.4.3 (because at the moment it is just a >= check ignoring patch versions). That might not be valid but at the same if they have a version 2 of shuttle service, will the server be running 0.4.3?

Reply:
Hmm... good point. Let's keep this for now then

Reply:
This is a good conversation. Have written this in a comment in the tests